### PR TITLE
HAWQ-1421. PXF rpm updates to make it oss friendly

### PR DESCRIPTION
**Pull request by shivzone**

### CODE REVIEW

File: pxf/build.gradle [hunk 1]
```diff
@@ -41,17 +41,22 @@ buildscript {
     }   
 }
 
-// Hadoop distribution (can be phd or hdp) - needed to set correct classpaths
+// Hadoop distribution (can be hdp or null which would default to apache) - needed to set correct classpaths
 // can be set with -Dhd=<value>
-def hddist = System.properties['hd'] ?: 'hdp';
-if (hddist != 'phd' && hddist != 'hdp') {
+def hddist = System.properties['hd'] ?: '';
```
Comment (sansanichfb) [inline pxf/build.gradle]: As long as we don't have default value, shouldn't it be just: def hddist = System.properties['hd']?

**Comment by radarwave** [comment]
After set "export HD=hdp", the rpm package names changed back with the version string "3_2_1_0". Maybe this is expected. But then I still seeing it requires Apache Hadoop. See:

`rpm -ivh pxf-service_3_2_1_0-3.2.1.0-1.el6.noarch.rpm
error: Failed dependencies:
	hadoop >= 2.7.1 is needed by pxf-service_3_2_1_0-0:3.2.1.0-1.el6.noarch
	hadoop-hdfs >= 2.7.1 is needed by pxf-service_3_2_1_0-0:3.2.1.0-1.el6.noarch`

File: pxf/build.gradle (hunk 1 above)
Comment (shivzone) [inline pxf/build.gradle]: System.properties returns null and not ''.

**Comment by shivzone** [comment]
Can you please provide me the output of sudo yum list installed | grep hadoop
If this was hdp distribution, hadoop-hdfs rpms should have already been installed or been available in some repo
[gpadmin@test1 ~]$ rpm -qR hadoop-hdfs.noarch
hadoop_2_5_0_0_1245-hdfs

**Comment by radarwave** [comment]
sudo yum list installed | grep hadoop
hadoop-conf-pseudo.noarch        2.7.3.2.5.0.0-1245.el6     @HDP-2.5.0.0
hadoop_2_5_0_0_1245.x86_64       2.7.3.2.5.0.0-1245.el6     @HDP-2.5.0.0
hadoop_2_5_0_0_1245-conf-pseudo.x86_64
hadoop_2_5_0_0_1245-hdfs.x86_64  2.7.3.2.5.0.0-1245.el6     @HDP-2.5.0.0
hadoop_2_5_0_0_1245-hdfs-datanode.x86_64
hadoop_2_5_0_0_1245-hdfs-namenode.x86_64
hadoop_2_5_0_0_1245-hdfs-secondarynamenode.x86_64
hadoop_2_5_0_0_1245-mapreduce.x86_64
hadoop_2_5_0_0_1245-mapreduce-historyserver.x86_64
hadoop_2_5_0_0_1245-yarn.x86_64  2.7.3.2.5.0.0-1245.el6     @HDP-2.5.0.0
hadoop_2_5_0_0_1245-yarn-nodemanager.x86_64
hadoop_2_5_0_0_1245-yarn-resourcemanager.x86_64

rpm -qR hadoop-hdfs.noarch
package hadoop-hdfs.noarch is not installed

**Comment by shivzone** [comment]
@radarwave looks like you have installed the HDP rpms without using the virtual rpms. Products that depend on hdp will only have a dependancy on their virtual rpms as it is close to impossible to keep track of the exact build number of hadoop as part of the dependancy.

hadoop virutal rpm will be hadoop-hdfs which will have a dependancy on the physical rpm hadoop_2_5_0_0_1245-hdfs. Hope this fixes your issue.

**Comment by radarwave** [comment]
Thanks @shivzone .The virtual packages works to me.

+1

**Comment by huor** [comment]
+1 for great effort to building hawq with apache hadoop

**Pull request closed without merging**